### DISCUSSION TITLE
Changing map height limit sandbox option reloads NewGRF data – is that still necessary?

### DISCUSSION THREAD

**Discussion by James103** [OpenTTD/OpenTTD#13033] [General]
### Version of OpenTTD
Since 7b27a28f0a961f23b2998b65a06723fcfc3a3535

### Expected result
Changing the map height limit should not require reloading NewGRF data, if it isn't necessary to do so.

### Actual result
Since 7b27a28f0a961f23b2998b65a06723fcfc3a3535, when the sandbox option to edit the maximum map height was introduced, changing it from the Sandbox Options automatically ran the console command `reload_newgrfs`, causing the game to temporarily hang as it reloaded NewGRF data.
https://github.com/OpenTTD/OpenTTD/blob/bb8a0c76413c5817cf54fec81e400ab4e8b50491/src/cheat_gui.cpp#L156-L158

However, using the Scenario Editor to change the same map height limit setting from Settings → Limitations → Map height limit does not automatically reload NewGRF data.

What mechanics used in-game, if any, require that NewGRF data is reloaded when the map height limit is changed?

### Steps to reproduce
* Open the Scenario Editor and change Settings → Limitations → Map height limit. This does not reload NewGRF data.
* Start a new game, open Sandbox Options, and change the maximum map height there. This reloads NewGRF data.

**Comment by JGRennison** [comment]
It's needed because of Action D patch variable 14, and to a lesser extent for recalculating snow line tables.
